src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PreReleaseContactsService.cs

```diff
@@ -0,0 +1,299 @@
+                                        CreatedBy = ""
+                                    });
+                                    await _usersAndRolesDbContext.SaveChangesAsync();
+                                }
+                            }
+                        }
+
+                        await SendPreReleaseInviteEmail(releaseId, email, newUser);
+                    }
+                })
+                .OnSuccess(_ => GetPreReleaseContactsForReleaseAsync(releaseId));
+        }
+
+
+        public async Task<Either<ActionResult, List<PrereleaseCandidateViewModel>>>
+            RemovePreReleaseContactFromReleaseAsync(
+                Guid releaseId, string email)
+        {
+            return await _persistenceHelper
+                .CheckEntityExists<Release>(releaseId)
+                .OnSuccess(_userService.CheckCanAssignPrereleaseContactsToRelease)
+                .OnSuccessDo(async () =>
+                {
+                    var existingUserRole = await _context
+                        .UserReleaseRoles
+                        .Include(r => r.User)
+                        .Where(r =>
+                            r.ReleaseId == releaseId
+                            && r.User.Email.ToLower() == email.ToLower()
+                            && r.Role == ReleaseRole.PrereleaseViewer)
+                        .FirstOrDefaultAsync();
+
+                    if (existingUserRole != null)
+                    {
+                        _context.Remove(existingUserRole);
+                        await _context.SaveChangesAsync();
+                    }
+                    else
+                    {
+                        var existingInviteToRelease = await _context
+                            .UserReleaseInvites
+                            .Where(i =>
+                                i.ReleaseId == releaseId
+                                && i.Email.ToLower() == email.ToLower()
+                                && i.Role == ReleaseRole.PrereleaseViewer)
+                            .FirstOrDefaultAsync();
+
+                        if (existingInviteToRelease != null)
+                        {
+                            // TODO - also need to remove their overall UserInvite record if they have no more 
+                            // UserReleaseInvites available
+                            _context.Remove(existingInviteToRelease);
+                            await _context.SaveChangesAsync();
+                        }
+                    }
+                })
+                .OnSuccess(_ => GetPreReleaseContactsForReleaseAsync(releaseId));
+        }
+
+        private async Task SendPreReleaseInviteEmail(Guid releaseId, string email, bool newUser)
+        {
+            var template = _configuration.GetValue<string>("NotifyPreReleaseTemplateId");
+
+            var release = await _context.Releases.Include(r => r.Publication)
+                .FirstOrDefaultAsync(r => r.Id == releaseId);
```

Comment:
I think you could use `.First` here since you don't later check if the release is null

Reply:
Done